packages/runtime/container-runtime/src/opLifecycle/outbox.ts

```diff
@@ -32,40 +36,62 @@ export interface IOutboxParameters {
 }
 
 export class Outbox {
+	private readonly mc: MonitoringContext;
 	private readonly attachFlowBatch: BatchManager;
 	private readonly mainBatch: BatchManager;
-	private readonly defaultAttachFlowSoftLimitInBytes = 64 * 1024;
+	private readonly defaultAttachFlowSoftLimitInBytes = 320 * 1024;
 
 	constructor(private readonly params: IOutboxParameters) {
+		this.mc = loggerToMonitoringContext(ChildLogger.create(params.logger, "Outbox"));
 		const isCompressionEnabled =
 			this.params.config.compressionOptions.minimumBatchSizeInBytes !==
 			Number.POSITIVE_INFINITY;
 		// We need to allow infinite size batches if we enable compression
 		const hardLimit = isCompressionEnabled ? Infinity : this.params.config.maxBatchSizeInBytes;
 		const softLimit = isCompressionEnabled ? Infinity : this.defaultAttachFlowSoftLimitInBytes;
 
-		this.attachFlowBatch = new BatchManager(
-			{
-				hardLimit,
-				softLimit,
-				enableOpReentryCheck: params.config.enableOpReentryCheck,
-			},
-			params.logger,
-		);
-		this.mainBatch = new BatchManager(
-			{
-				hardLimit,
-				enableOpReentryCheck: params.config.enableOpReentryCheck,
-			},
-			params.logger,
-		);
+		this.attachFlowBatch = new BatchManager({ hardLimit, softLimit });
+		this.mainBatch = new BatchManager({ hardLimit });
 	}
 
 	public get isEmpty(): boolean {
 		return this.attachFlowBatch.length === 0 && this.mainBatch.length === 0;
 	}
 
+	/**
+	 * If we detect that the reference sequence number of the incoming message does not match
+	 * what was already in the batch manager, this means that the batch has been interrupted so
+	 * we will flush the accumulated messages to account for that and create a new batch with the new
+	 * message as the first message.
+	 *
+	 * @param batchManager - the batch manager where the message is supposed to be pushed to
+	 * @param message - the incoming message
+	 */
+	private maybeFlushPartialBatch(batchManager: BatchManager, message: BatchMessage) {
+		if (this.mc.config.getBoolean("Fluid.ContainerRuntime.DisablePartialFlush") === true) {
+			return;
+		}
+
+		const batchReference = batchManager.referenceSequenceNumber;
+		if (batchReference !== undefined && batchReference !== message.referenceSequenceNumber) {
+			this.mc.logger.sendErrorEvent(
+				{
+					eventName: "ReferenceSequenceNumberMismatch",
+					referenceSequenceNumber: batchReference,
+					messageReferenceSequenceNumber: message.referenceSequenceNumber,
+					type: message.deserializedContent.type,
+					count: batchManager.length,
+					batchSize: batchManager.contentSizeInBytes,
+				},
+				new UsageError("Submission of an out of order message"),
```

Comment:
I'd not add error object here - it has no purpose. Event name is all we care about.

Reply:
it was not really useful in production but during testing, the stack trace from this error definitely helps.